api/src/main/java/org/openmrs/module/patientqueueing/api/impl/PatientQueueingServiceImpl.java

```diff
@@ -78,4 +83,79 @@ public PatientQueue getIncompletePatientQueue(Patient patient, Location location
 		return dao.getIncompletePatientQueue(patient, locationTo);
 	}
 	
+	/**
+	 * This Method checks if a patient queueNumber exists
+	 * 
+	 * @param queueNumber A queueNumber in form of a string which will be check to determine if it
+	 *            exists
+	 * @return
+	 */
+	private boolean isQueueNumberIdExisting(String queueNumber) {
+		
+		return getPatientQueueByQueueNumber(queueNumber) != null;
+	}
+
+	/**
+	 * @see org.openmrs.module.patientqueueing.api.PatientQueueingService#getPatientQueueByQueueNumber(java.lang.String)
+	 */
+	@Override
+	public PatientQueue getPatientQueueByQueueNumber(String queueNumber) {
+		List<PatientQueue> patientQueueList = dao.getPatientQueueByQueueNumber(queueNumber);
+		if (!patientQueueList.isEmpty()) {
+			return patientQueueList.get(0);
+		} else {
+			return null;
+		}
+
+	}
+
+	/**
+	 * @see org.openmrs.module.patientqueueing.api.PatientQueueingService#generateQueueNumber(org.openmrs.Location,
+	 *      java.util.Date, org.openmrs.Patient)
+	 */
+	public String generateQueueNumber(Location location, Date date, Patient patient) {
+
+		List<PatientQueue> patientQueueList = new ArrayList<PatientQueue>();
+
+		try {
+			patientQueueList = getPatientQueueList(null, QueueingUtils.convertDateToDefaultDateFormat(date, "00:00:00"),
+					QueueingUtils.convertDateToDefaultDateFormat(date, "23:59:59"), null, location, patient, null);
```

Comment:
What is the use of this magic text? `23:59:59`

Reply:
Beginning of day and end of day.

Reply:
Can you codify that? Such that one does not have to look up these github comments to understand what that means?